radixdlt-core/radixdlt/src/test/java/com/radixdlt/stateir/StateIRSerializationTest.java

```diff
@@ -0,0 +1,282 @@
+import com.radixdlt.statecomputer.LedgerAndBFTProof;
+import com.radixdlt.statecomputer.checkpoint.MockedGenesisModule;
+import com.radixdlt.statecomputer.forks.Forks;
+import com.radixdlt.statecomputer.forks.ForksModule;
+import com.radixdlt.statecomputer.forks.RERulesConfig;
+import com.radixdlt.statecomputer.forks.RadixEngineForksLatestOnlyModule;
+import com.radixdlt.statecomputer.forks.modules.MainnetForksModule;
+import com.radixdlt.store.DatabaseLocation;
+import com.radixdlt.store.LastStoredProof;
+import com.radixdlt.store.berkeley.BerkeleyLedgerEntryStore;
+import com.radixdlt.utils.PrivateKeys;
+import com.radixdlt.utils.UInt256;
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.IntStream;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public final class StateIRSerializationTest {
+  private static final ECKeyPair VALIDATOR_KEY = PrivateKeys.ofNumeric(1);
+
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  @Inject private RadixEngine<LedgerAndBFTProof> radixEngine;
+
+  @Inject private BerkeleyLedgerEntryStore ledgerEntryStore;
+
+  // FIXME: Hack, need this in order to cause provider for genesis to be stored
+  @Inject @LastStoredProof private LedgerProof ledgerProof;
+
+  @Inject private Forks forks;
+
+  private final Random rnd = new Random();
+
+  private Injector createInjector() {
+    return Guice.createInjector(
+        MempoolConfig.asModule(1000, 10),
+        new MainnetForksModule(),
+        new RadixEngineForksLatestOnlyModule(
+            RERulesConfig.testingDefault()
+                .overrideFeeTable(
+                    FeeTable.create(
+                        Amount.zero(), Map.of(TokenResource.class, Amount.ofTokens(1))))),
+        new ForksModule(),
+        new SingleNodeAndPeersDeterministicNetworkModule(VALIDATOR_KEY, 0),
+        new MockedGenesisModule(
+            Set.of(VALIDATOR_KEY.getPublicKey()),
+            Amount.ofTokens(100_000_000_000L),
+            Amount.ofTokens(100L)),
+        new AbstractModule() {
+          @Override
+          protected void configure() {
+            bindConstant()
+                .annotatedWith(DatabaseLocation.class)
+                .to(folder.getRoot().getAbsolutePath());
+          }
+        });
+  }
+
+  @Test
+  public void
+      it_should_successfully_construct_and_then_serialize_and_deserialize_the_intermediate_state()
+          throws Exception {
+    createInjector().injectMembers(this);
+
+    final var numAccounts = 100;
+    final var numTokens = 10;
+    final var transfersPerToken = 100;
+
+    final var accounts =
+        IntStream.range(0, numAccounts)
+            .mapToObj(unused -> REAddr.ofPubKeyAccount(ECKeyPair.generateNew().getPublicKey()))
+            .toList();
```

Comment:
I think it is a good idea to try to extract the test preparation to private methods which abstract away the noise ;).Ffor example: 

`final var accounts = getAccounts(numAccounts);`